src/types/composite/container.ts

```diff
@@ -26,16 +26,30 @@ export const CONTAINER_TYPE = Symbol.for("ssz/ContainerType");
 export function isContainerType<T extends ObjectLike = ObjectLike>(type: Type<unknown>): type is ContainerType<T> {
   return isTypeOf(type, CONTAINER_TYPE);
 }
+type FieldInfo = {
+  isBasic: boolean;
+  gindex: bigint;
+};
 
 export class ContainerType<T extends ObjectLike = ObjectLike> extends CompositeType<T> {
   // ES6 ensures key order is chronological
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   fields: Record<string, Type<any>>;
+  fieldInfos: Map<string, FieldInfo>;
```

Comment:
Can you add a nice JSDoc comment explaining the improvement with this cache, the expected speed up, etc?